plugins/kernel_v2/config/vm.rb

```diff
@@ -748,6 +804,21 @@ def validate(machine, ignore_provider=nil)
           end
         end
 
+        # Validate disks
+        # Check if there is more than one primrary disk defined and throw an error
+        primary_disks = @disks.select { |d| d.primary && d.type == :disk }
+        if primary_disks.size > 1
+          errors << I18n.t("vagrant.config.vm.multiple_primary_disks_error",
+                           name: machine.name)
+        end
+
+        # TODO: Check for duplicate disk names?
```

Comment:
Yeah, even though we have an ID we should still keep the disk names unique for clarity